missioncontrol/home/migrations/0021_s3files_20190227_0220.py

```diff
@@ -0,0 +1,48 @@
+# Generated by Django 2.1.4 on 2019-02-27 03:37
+
+from django.db import migrations, models
+import django.db.models.deletion
+import django.utils.timezone
+import home.models
+import uuid
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('home', '0020_merge_20190124_0028'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='S3File',
+            fields=[
```

Comment:
This is the schema I'd recommend.
Have a read through : https://github.com/planetlabs/datalake-common
I think it's quite well thought out, although I'd:
1. replace the id/hash with a CID
2. move from dynamodb to postgres
3. use iso8601 z-dot
 ```
  {
        "version": 0,
        "start": <iso-8601.z>,
        "end": <iso-8601.z>,
        "path": "/var/log/syslog.1"
        "work_id": null,
        "where": "webserver02",
        "what": "syslog",
        "cid": "6309e115c2914d0f8622973422626954",
    }
```

Reply:
Added path, start and end, kept created though. 'work_id' is 'task_run' in this case (we can change this later if required?)

Didn't see a need for version, as django migrations will handle any changes to this schema.

Reply:
`version` is not the schema version, it's the version of the file.

Deleting files should only be done carefully / as an administrative task, so you need a way to update them. You can serve up the max file version by default.

`created` is always a good addition :)

I'd like to push back on `task_run`/ `work_id`, as it allows us to use this model for many more things.

I don't think there should be a direct relationship between the file model and any other models. There's significant benefit to introducing one layer of indirection there.

The idea is that between the combination of `what` and `work_id` you can look resolve files from multiple services or jobs, (assuming you namespace appropriately). The reason i'd want to push on this is that you get massive organizational benefit by having a shared one-true-datalake. However if this model becomes task_run specific (or even mc-specific), then data storage will fragment into 100 buckets (that's the hard limit on buckets per AWS account btw.. :scream_cat: ).

For example, I'd love to use the files endpoint for storing rotated logs, telemetry, from every cron job and service being used. Not all of those things will have models in mc. It would be really helpful if you could easily provision a `what` like "jobs-gs-scheduler-stdout" where the `work_id` could be a jenkins job id (for example).

I also think there's real benefit of being able to leverage datalake as a microservice down the line. I'd want to avoid creating something really similar, but different enough that they can't be made compatible with a few PRs to upstream (i.e. why would they merge something about task_runs, or delete the version field?).

Reply:
Added version.
Renamed 'task_run' foreign key to a generic 'work_id' (which should be a UUID)